src/containers/quiz/quiz-question/Question.tsx

```diff
@@ -0,0 +1,158 @@
+import { ChangeEventHandler, FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Box, Typography, FormGroup, RadioGroup, SxProps } from '@mui/material'
+import { questionType } from '~/components/question-editor/QuestionEditor.constants'
+import AppCard from '~/components/app-card/AppCard'
+import CheckIcon from '@mui/icons-material/Check'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { Question } from '~/types'
+import { styles } from './Question.styles'
+import { isCorrectAnswer } from './Question.constants'
+import Answer from '../question-answer/Answer'
+import { spliceSx } from '~/utils/helper-functions'
+
+interface QuizQuestionProps {
+  question: Question
+  index: number
+  value: string | null | string[]
+  showCorrectAnswers?: boolean
+  showPoints?: boolean
+  showAnswersCorrectness?: boolean
+  isEditable?: boolean
+  useAppCard?: boolean
+  sx?: SxProps
+  handleInputChange: ChangeEventHandler
+  handleNonInputValueChange: (value: string | string[]) => void
+}
+
+const QuizQuestion: FC<QuizQuestionProps> = ({
+  useAppCard,
+  sx,
+  index,
+  showPoints,
+  showCorrectAnswers,
+  showAnswersCorrectness,
+  value,
+  question,
+  isEditable,
+  handleInputChange,
+  handleNonInputValueChange
+}) => {
+  const { t } = useTranslation()
+
+  const { isMultipleChoice, isSingleChoice, isOpenAnswer } = questionType(
+    question.type
+  )
+
+  const ContainerComponent = useAppCard ? AppCard : Box
+
+  const isCorrect = isCorrectAnswer(question, value)
+
+  const iconStyles = styles.icon(isCorrect ? 'correct' : 'incorrect')
+
+  const correctnessIcon =
+    showAnswersCorrectness &&
+    (isCorrect ? <CheckIcon sx={iconStyles} /> : <CloseIcon sx={iconStyles} />)
+
+  const shouldShowCorrectAnswers = showCorrectAnswers && !isOpenAnswer
+  const correctAnswers = shouldShowCorrectAnswers && (
+    <Box sx={styles.correctAnswers.root}>
+      <Typography sx={styles.correctAnswers.title}>
+        {t('myResourcesPage.quizzes.correctAnswers')}
+      </Typography>
+      <Box sx={styles.correctAnswers.list}>
+        {question.answers
+          .filter((item) => item.isCorrect)
+          .map((item) => (
```

Comment:
i think would be better to move it up to separate variable

Reply:
Thanks, moved it to separate variable!